homeassistant/components/api.py

```diff
@@ -79,6 +80,7 @@ def get(self, request):
         if restrict:
             restrict = restrict.split(',') + [EVENT_HOMEASSISTANT_STOP]
 
+        @asyncio.coroutine
```

Comment:
Components are setup outside of the event loop, so the normal listen is used to schedule the call. Coroutine check is done when triggering listeners.